src/hera/resources.py

```diff
@@ -70,15 +70,20 @@ def __post_init__(self):
             if self.cpu_limit is not None and isinstance(self.cpu_limit, int):
                 assert self.cpu_request <= self.cpu_limit, "CPU request must be smaller or equal to limit"
 
+        if self.cpu_request is None and self.cpu_limit is not None:
+            self.cpu_request = self.cpu_limit
+        if self.memory_request is None and self.memory_limit is not None:
+            self.memory_request = self.memory_limit
+
     def build(self) -> ResourceRequirements:
         """Builds the resource requirements of the pod"""
         resources = dict()
 
         if self.cpu_limit is not None:
-            resources = merge_dicts(resources, dict(limit=dict(cpu=self.cpu_limit)))
+            resources = merge_dicts(resources, dict(limit=dict(cpus=str(self.cpu_limit))))
```

Comment:
Should this one be `cpus` plural (but singular below for request)?

Reply:
Independent of anything else yes, I agree with you, but recall that `ResourceRequirements` is mapped to Kubernetes resource requirements, and those use `cpu` only, inside the `requests` and `limits` fields! See [docs here](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#resource-units-in-kubernetes)!

Reply:
In that case, should this be:
```suggestion
            resources = merge_dicts(resources, dict(limit=dict(cpu=str(self.cpu_limit))))
```
?

Reply:
Oh... yes 😬